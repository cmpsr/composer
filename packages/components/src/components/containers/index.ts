export * from './Box';
export * from './Grid';
export * from './Section';
export * from './SectionChild';
