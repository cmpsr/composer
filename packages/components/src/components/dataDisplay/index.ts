export { ListItem } from '@chakra-ui/react';
export * from './Badge';
export * from './Divider';
export * from './NumberedList';
export * from './Symbol';
export * from './Tag';
export * from './UnorderedList';
