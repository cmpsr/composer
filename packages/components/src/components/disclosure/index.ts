export * from './Accordion';
export * from './Tabs';
