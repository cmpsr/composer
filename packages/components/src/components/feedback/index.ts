export * from './Alert';
export * from './InlineAlert';
export * from './ProgressBar';
export * from './ProgressCircular';
export * from './Skeleton';
export * from './Spinner';
export * from './Toast';
