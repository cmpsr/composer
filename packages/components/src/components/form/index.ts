export * from './Autocomplete';
export * from './AutocompleteMultiSelect';
export * from './Button';
export * from './ButtonIcon';
export * from './Checkbox';
export * from './DateInput';
export * from './IconButton';
export * from './Input';
export * from './MaskInput';
export * from './NumberInput';
export * from './PhoneInput';
export * from './PinInput';
export * from './Radio';
export * from './RangeSlider';
export * from './SegmentedButton';
export * from './Select';
export * from './Slider';
export * from './Switch';
export * from './Textarea';
