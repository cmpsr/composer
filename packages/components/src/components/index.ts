export * from './layouts';
export * from './primitives';
export * from './navigation';
export * from './notifications';
