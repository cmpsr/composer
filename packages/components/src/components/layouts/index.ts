export * from './Box';
export * from './Flex';
export * from './Grid';
export * from './Section';
