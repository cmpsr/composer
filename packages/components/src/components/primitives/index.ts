export * from './Accordion';
export * from './Alert';
export * from './Avatar';
export * from './Badge';
export * from './Breadcrumb';
export * from './Button';
export * from './Card';
export * from './CircularProgress';
export * from './Checkbox';
export * from './Divider';
export * from './Image';
export * from './Icons';
export * from './Input';
export * from './Link';
export * from './Progress';
export * from './Radio';
export * from './Select';
export * from './Slider';
export * from './Spinner';
export * from './Switch';
export * from './Tabs';
export * from './Tag';
export * from './Text';
export * from './TextPairing';
export * from './Textarea'
