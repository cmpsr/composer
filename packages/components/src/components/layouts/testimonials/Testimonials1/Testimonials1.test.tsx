import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Testimonials1, TESTIMONIAL1_PLAN_DEFAULT_TEST_ID } from '.';
import { IMAGE_DEFAULT_TEST_ID } from 'components/primitives';

describe('Testimonials1', () => {
  const asset = {
    contentType: 'image',
    title: 'Image Title',
    url:
      'https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg',
  };
  const testimonial =
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae aut, omnis, maxime rem, cum similique repellendus hic impedit totam magnam deserunt voluptatum quia! Blanditiis repellendus sit debitis doloribus, totam minima?';
  const author = 'Testing author,';
  const profession = 'Testing profession.';

  it('should match snapshot', () => {
    const testimonial1 = renderer.create(
      <Testimonials1
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );

    const tree = testimonial1.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render component', () => {
    const { getByTestId } = render(
      <Testimonials1
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial1 = getByTestId(TESTIMONIAL1_PLAN_DEFAULT_TEST_ID);
    expect(testimonial1).toBeInTheDocument();
  });
  it('should render asset', () => {
    const { getByTestId } = render(
      <Testimonials1
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );

    const assetComponent = getByTestId(IMAGE_DEFAULT_TEST_ID);
    expect(assetComponent).toBeInTheDocument();
  });
  it('should render testimonial', () => {
    const { getByText } = render(
      <Testimonials1
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial1 = getByText(testimonial);
    expect(testimonial1).toBeInTheDocument();
  });
  it('should render author', () => {
    const { getByText } = render(
      <Testimonials1
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial1 = getByText(author);
    expect(testimonial1).toBeInTheDocument();
  });
  it('should render profession', () => {
    const { getByText } = render(
      <Testimonials1
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial1 = getByText(profession);
    expect(testimonial1).toBeInTheDocument();
  });
});
