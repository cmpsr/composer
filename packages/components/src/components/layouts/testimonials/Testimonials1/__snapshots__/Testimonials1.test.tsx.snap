// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials1 should match snapshot 1`] = `
<div
  className="w-full h-full max-w-content mx-auto flex items-stretch"
  data-testid="box"
>
  <div
    className="flex flex-col md:flex-row ml-8 mr-4 my-8 md:my-20 md:mx-24"
    data-testid="testimonial1"
  >
    <div
      className="w-40 h-40 md:w-64 md:h-64 md:mr-12"
    >
      <picture
        className="select-none"
      >
        <source
          className="select-none h-full object-cover"
          data-testid="image"
          srcSet="https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg?fm=webp"
          type="image/webp"
        />
        <img
          alt="Image Title"
          className="select-none h-full object-cover"
          src="https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg"
        />
      </picture>
    </div>
    <div
      className="flex flex-col self-center text-left md:flex-row md:flex-1"
    >
      <div
        className="mt-3 mr-4 md:mt-0"
      >
        <svg
          data-testid="icon"
          height="50px"
          viewBox="0 0 24 24"
          width="50px"
        >
          <path
            d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
          />
        </svg>
      </div>
      <div
        className="pt-3"
      >
        <p
          className="typo-headline-5 font-bold"
          data-testid="typography"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae aut, omnis, maxime rem, cum similique repellendus hic impedit totam magnam deserunt voluptatum quia! Blanditiis repellendus sit debitis doloribus, totam minima?
        </p>
        <div
          className="mt-4"
        >
          <span
            className="typo-body mr-2"
            data-testid="typography"
          >
            Testing author,
          </span>
          <span
            className="typo-body text-gray-500"
            data-testid="typography"
          >
            Testing profession.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
